common/tasks/interleaved_weighted_round_robin.go

```diff
@@ -231,9 +237,22 @@ func (s *InterleavedWeightedRoundRobinScheduler[T, K]) getOrCreateTaskChannel(
 
 func (s *InterleavedWeightedRoundRobinScheduler[T, K]) flattenWeightedChannelsLocked() {
 	weightedChannels := make(WeightedChannels[T], 0, len(s.weightedChannels))
-	for _, weightedChan := range s.weightedChannels {
+	var keysToDelete []K
+	for k, weightedChan := range s.weightedChannels {
+		if s.options.InactiveChannelDeletionDelay != nil &&
+			s.ts.Now().Sub(weightedChan.LastActiveTime()) > s.options.InactiveChannelDeletionDelay() &&
```

Comment:
nit: let's get the DeletionDelay only once.

Reply:
Done